### ISSUE TITLE
Handle conditionally active mods in LO

### ISSUE THREAD

**Issue by ryan-rushton** [DestinyItemManager/DIM#7509]
### Proposed change

We need to handle conditionally active mods in LO that enhance stats. These are
1. Radiant light
2. Powerful friends
3. Charged harvester

When selecting these mods we should be adding their conditional stat benefit/deficit to the overall set stats.

**Comment by smerrill** [comment]
Hey - I was thinking of taking this on as a good first issue. Looks like this partially works at the moment? I see `isModStatActive` in loadout-utils.ts, but at the same time, I see that Powerful Friends currently works (perhaps since I have another Arc mod slotted for no reason in my PvP build) but Radiant Light does not.

<img width="471" alt="image" src="https://user-images.githubusercontent.com/156815/148303957-c8c547d1-db8e-4dc8-995c-3db429a2d22e.png">
<img width="472" alt="image" src="https://user-images.githubusercontent.com/156815/148304000-cdade645-9bf4-42d0-9553-fc8ce97e5ed5.png">

**Comment by smerrill** [comment]
Nevermind; it appears that `isPlugStatActive` is never called with its final optional `modsOnOtherItems?` param, so this is perhaps a separate issue.

**Comment by ryan-rushton** [comment]
Yeah so this feature is specifically in the Loadout Optimizer. When we map items we are partially handling them https://github.com/DestinyItemManager/DIM/blob/master/src/app/loadout-builder/process/mappers.ts#L64-L70. The issue with this we are only looking for other changed with light mods. 

There are other rules to activate these mods too, from memory its having another arc mod slotted in the same item. The hard part about this is we can't pre-calculate it like we are doing at the moment. We would need to track whether we have achieved the condition when we are assessing whether mods can fit onto an item set. This is hard as it would slow down the Optimizer processing significantly if we did this for every set me validate.
